debug,backup: crash when inspecting backup stored on s3

**Describe the problem**

When running `cockroach debug backup {show,list-backups,list-incremental,export} 's3://path/to/my/backup`, I see:

```
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1  a panic has occurred!
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +runtime error: invalid memory address or nil pointer dereference
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +(1) attached stack trace
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  -- stack trace:
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | runtime.sigpanic
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/usr/local/go/src/runtime/signal_unix.go:734
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cloud/amazon.MakeS3Storage
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cloud/amazon/s3_storage.go:255
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cloud.MakeExternalStorage
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cloud/impl_registry.go:153
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cloud.ExternalStorageFromURI
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cloud/impl_registry.go:95
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/ccl/cliccl.externalStorageFromURIFactory
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/cliccl/debug_backup.go:283
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.ReadBackupManifestFromURI
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/manifest_handling.go:125
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/ccl/cliccl.getManifestFromURI
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/cliccl/debug_backup.go:296
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/ccl/cliccl.runExportDataCmd
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/cliccl/debug_backup.go:418
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:68
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/spf13/cobra.(*Command).execute
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:852
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/spf13/cobra.(*Command).ExecuteC
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:960
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/spf13/cobra.(*Command).Execute
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:897
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.Run
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:295
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.doMain
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:127
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | github.com/cockroachdb/cockroach/pkg/cli.Main
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:61
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | main.main
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/cockroach/main.go:26
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | runtime.main
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/usr/local/go/src/runtime/proc.go:225
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | runtime.goexit
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +Wraps: (2) runtime error: invalid memory address or nil pointer dereference
E210817 21:20:01.071839 1 1@util/log/logcrash/crash_reporting.go:140  [-] 1 +Error types: (1) *withstack.withStack (2) runtime.errorString
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7a65e9f]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.RecoverAndReportPanic(0xa10d6a0, 0xc000138060, 0xc000b92000)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:90 +0xa5
panic(0x83a5600, 0xcdba9f0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/cockroachdb/cockroach/pkg/cloud/amazon.MakeS3Storage(0xa10d6a0, 0xc000138060, 0x0, 0x0, 0x9194140, 0x0, 0x0, 0x0, 0x3, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/cloud/amazon/s3_storage.go:255 +0x27f
github.com/cockroachdb/cockroach/pkg/cloud.MakeExternalStorage(0xa10d6a0, 0xc000138060, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0011aafa0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/cloud/impl_registry.go:153 +0x195
github.com/cockroachdb/cockroach/pkg/cloud.ExternalStorageFromURI(0xa10d6a0, 0xc000138060, 0x7ffeefbffb16, 0x79, 0x4000000, 0x0, 0x9194140, 0x8acdf91, 0x4, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/cloud/impl_registry.go:95 +0x235
github.com/cockroachdb/cockroach/pkg/ccl/cliccl.externalStorageFromURIFactory(0xa10d6a0, 0xc000138060, 0x7ffeefbffb16, 0x79, 0x8acdf91, 0x4, 0x8, 0xde7, 0x3266695a54, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/cliccl/debug_backup.go:283 +0xb6
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.ReadBackupManifestFromURI(0xa10d6a0, 0xc000138060, 0x7ffeefbffb16, 0x79, 0x8acdf91, 0x4, 0x9194118, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/manifest_handling.go:125 +0xd5
github.com/cockroachdb/cockroach/pkg/ccl/cliccl.getManifestFromURI(0xa10d6a0, 0xc000138060, 0x7ffeefbffb16, 0x79, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/cliccl/debug_backup.go:296 +0x145
github.com/cockroachdb/cockroach/pkg/ccl/cliccl.runExportDataCmd(0xc00065c280, 0xc0012fe960, 0x1, 0x2, 0x10a988a027f39d2a, 0x10)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/cliccl/debug_backup.go:418 +0x17a
github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1(0xc00065c280, 0xc0012fe960, 0x1, 0x2, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:68 +0x79
github.com/spf13/cobra.(*Command).execute(0xc00065c280, 0xc0012fe940, 0x2, 0x2, 0xc00065c280, 0xc0012fe940)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xce03560, 0xc000138060, 0xc000ade08a, 0x13)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:897
github.com/cockroachdb/cockroach/pkg/cli.Run(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:295
github.com/cockroachdb/cockroach/pkg/cli.doMain(0xc00065c280, 0xc000ade08a, 0x13, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:127 +0x11a
github.com/cockroachdb/cockroach/pkg/cli.Main()
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:61 +0x13a
main.main()
	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/cockroach/main.go:26 +0x25
```

**To Reproduce**

See above.


**Expected behavior**
Not to panic.


**Environment:**
```
Build Tag:        v21.2.0-alpha.00000000-3004-ge0c8f31fb6
Build Time:       2021/08/04 16:37:09
Distribution:     CCL
Platform:         darwin amd64 (x86_64-apple-darwin19)
Go Version:       go1.16.6
C Compiler:       Clang 10.0.0 
Build Commit ID:  e0c8f31fb652c19d0c04bc1b69f4d6de35f6687f
Build Type:       release
```

--- a/pkg/ccl/cliccl/debug_backup.go
+++ b/pkg/ccl/cliccl/debug_backup.go
@@ -281,8 +281,10 @@ func newBlobFactory(ctx context.Context, dialing roachpb.NodeID) (blobs.BlobClie
 func externalStorageFromURIFactory(
 	ctx context.Context, uri string, user security.SQLUsername,
 ) (cloud.ExternalStorage, error) {
+	defaultSettings := &cluster.Settings{}
+	defaultSettings.SV.Init(ctx, nil /* opaque */)
 	return cloud.ExternalStorageFromURI(ctx, uri, base.ExternalIODirConfig{},
-		cluster.NoSettings, newBlobFactory, user, nil /*Internal Executor*/, nil /*kvDB*/)
+		defaultSettings, newBlobFactory, user, nil /*Internal Executor*/, nil /*kvDB*/)
 }
 
 func getManifestFromURI(ctx context.Context, path string) (backupccl.BackupManifest, error) {

